### PR TITLE
TM-44 Localization

### PR DESCRIPTION
**Pull request by TomsBemis**
- Installed ngx-translate package
 - Localization JSON files for english and latvian, labels, roles, task statuses and types are using translate module
 - Dropdown for switching between languages
 - Submit buttons and nav items float to the end or start correctly
 
image
image

### CODE REVIEW

File: src/app/shared/translate/translate.module.ts [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { NgModule } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { TranslateHttpLoader } from '@ngx-translate/http-loader';
+import { HttpClientModule, HttpClient } from '@angular/common/http';
+import { TranslateModule, TranslateStore, TranslateService } from '@ngx-translate/core';
+import { AppComponent } from '../../app.component';
+
+export function HttpLoaderFactory(http: HttpClient): TranslateHttpLoader {
+  return new TranslateHttpLoader(http, '../../assets/i18n/','.json');
+
```
Comment (wanpii) [inline src/app/shared/translate/translate.module.ts]: do you really need a separate module for translation config?

File: src/app/app.component.ts [hunk 2]
```diff
@@ -1,18 +1,25 @@
...
+    TranslateModule
   ],
   templateUrl: './app.component.html',
   styleUrl: './app.component.scss'
 })
 export class AppComponent {
   title = 'task-manager';
 
+  constructor(private translate: TranslateService) {
+    this.translate.addLangs(['en', 'lv']);
```
Comment (wanpii) [inline src/app/app.component.ts]: you could move all of the translation logic to a separate service maybe, to cleanup code and do all the initial setting etc on the service?

File: src/app/shared/translate/translate.module.ts (hunk 1 above)
Comment (TomsBemis) [reply src/app/shared/translate/translate.module.ts]: How do I provide a translate loader to the app? It needs the http client as a parameter.
export const appConfig: ApplicationConfig = {
  providers: [
    importProvidersFrom(
      HttpClientModule
    ),
    TranslateService,
    TranslateStore,
    TranslateModule.forRoot({
      loader: {
        provide: TranslateLoader,
        useFactory: HttpLoaderFactory,
        deps: [HttpClient]
      }
    }).providers,
    provideRouter(appRoutes), 
    provideClientHydration(),
    tokenInterceptorProvider,
  ]
};
When I try to provide @ngx-translate/core TranslateModule with the loader I get an error:
`Type '(Provider | EnvironmentProviders)[] | undefined' is not assignable to type 'Provider | EnvironmentProviders'.
  Type 'undefined' is not assignable to type 'Provider | EnvironmentProviders'.`

Comment (wanpii) [reply src/app/shared/translate/translate.module.ts]: try importProvidersFrom(
TranslateModule.forRoot({etc. etc.})
)

**Review by wanpii — APPROVED** [review]

**Pull request merged**